appium-tests/helpers/cameraHelper.js

```diff
@@ -103,11 +109,185 @@ module.exports.getPicture = function (opts, pid) {
     }, opts);
 };
 
-module.exports.checkPicture = function (pid, cb) {
+// verifies taken picture when the promise is resolved,
+// calls a callback with 'OK' if everything is good,
+// calls a callback with 'ERROR: <error message>' if something is wrong
+// note that this function is executed in the context of tested app
+// and not in the context of tests
+module.exports.checkPicture = function (pid, options, cb) {
+    var isIos = cordova.platformId === "ios";
+    var isAndroid = cordova.platformId === "android";
+    // skip image type check if it's unmodified on Android:
+    // https://github.com/apache/cordova-plugin-camera/#android-quirks-1
+    var skipFileTypeCheck = isAndroid &&
+        (!options.quality || options.quality === 100) &&
```

Comment:
@riknoll actually if I remove that check, some specs with default quality are failing. For example, this options: `sourceType: PHOTOLIBRARY, destinationType: FILE_URI, encodingType: PNG, allowEdit: false, correctOrientation: false` return JPEG file.

Is this bug, so I should leave this check here?